### PR TITLE
Add `normalization_resolver` to resolve normalization layers

### PR DESCRIPTION
**Pull request by lightaime**
This pull request adds the `normalization_resolver` to support resolving normalization layers, which provides a flexible way to choose normalization layers in GNNs.

### CODE REVIEW

**Review by rusty1s — APPROVED** [review]
Thank you!

File: test/nn/test_resolver.py [hunk 1]
```diff
@@ -34,3 +35,28 @@ def test_aggregation_resolver(aggr_tuple):
     aggr_module, aggr_repr = aggr_tuple
     assert isinstance(aggregation_resolver(aggr_module()), aggr_module)
     assert isinstance(aggregation_resolver(aggr_repr), aggr_module)
+
+
+@pytest.mark.parametrize('norm_tuple', [
+    (torch_geometric.nn.norm.BatchNorm, 'batchnorm'),
```
[the remark this answers is not in the document]
Comment (lightaime) [reply test/nn/test_resolver.py]: How do you like to test against them? Do you mean to test their forward passes to check if they have different normalization results, etc?

[the remark this answers is not in the document]
Comment (rusty1s) [reply test/nn/test_resolver.py]: No, I mean that we can resolve both `batchnorm` and `batch_norm`.

[the remark this answers is not in the document]
Comment (lightaime) [reply test/nn/test_resolver.py]: I see. The `normalize_string` function should take care of this already: 
https://github.com/pyg-team/pytorch_geometric/blob/e02bd10aecae61c744ce1573dba526c8fba97fc0/torch_geometric/nn/resolver.py#L8
Do we need to add tests both of them?

[the remark this answers is not in the document]
Comment (rusty1s) [reply test/nn/test_resolver.py]: You are right. In that case, let‘s just mix the names up, e.g., use batchnorm for batch normalization and layer_norm for layer normalization.
